discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -220,6 +220,28 @@ impl SqliteStorage {
         Ok(())
     }
 
+    async fn store_new_search_documents(
+        tx: &mut Transaction<'_, Sqlite>,
+        documents: &[NewDocument],
+    ) -> Result<(), Error> {
+        if documents.is_empty() {
+            return Ok(());
+        }
+
+        SqliteStorage::store_new_documents(tx, documents).await?;
+
+        QueryBuilder::new("INSERT INTO SearchDocument (documentId) ")
+            .push_values(documents, |mut stm, doc| {
+                stm.push_bind(&doc.id);
+            })
+            .build()
+            .persistent(false)
+            .execute(tx)
+            .await?;
+
+        Ok(())
+    }
+
```

Comment:
Code was de-duplicated the SQL is the same but the `documents.is_empty()` check was added.